lib/translit.rb

```diff
@@ -32,10 +31,21 @@ def self.convert(text, enforce_language = nil)
 
   def self.invert_character_map(map)
     map.dup.inject({}) do |acc, tuple|
+      rus = tuple.last.first
+      eng_value       = tuple.first
+      acc[rus] ? acc[rus] << eng_value            : acc[rus] = [eng_value]
+      acc
+    end
+  end
+
+  def self.latin_cases(map)
+    map.dup.inject({}) do |acc, tuple|
       rus_up, rus_low = tuple.last
       eng_value       = tuple.first
-      acc[rus_up]  ? acc[rus_up]  << eng_value.capitalize : acc[rus_up]  = [eng_value.capitalize]
-      acc[rus_low] ? acc[rus_low] << eng_value            : acc[rus_low] = [eng_value]
+      acc[eng_value] = [rus_low]
+      unless eng_value == eng_value.capitalize
+        acc[eng_value.capitalize] = [rus_up]
+      end
       acc
     end
   end
```

Comment:
This new method now takes the hashes we've defined below with just the lower case latin characters and converts them to hashes with both the lower and upper case latin characters as keys. The values are the corresponding upper or lower case cyrillic letter.